core/common/src/main/java/alluxio/underfs/UnderFileSystem.java

```diff
@@ -614,6 +615,48 @@ default Fingerprint getParsedFingerprint(String path) {
   @Nullable
   UfsStatus[] listStatus(String path, ListOptions options) throws IOException;
 
+  /**
+   * Returns an array of statuses of the files and directories in the directory denoted by this
+   * abstract pathname, with options.
+   *
+   * <p>
+   * If this abstract pathname does not denote a directory, meaning it's a file, then this method
+   * would try to call getStatus on the path.
+   * Otherwise an array of statuses is returned, one for each file or directory. Names denoting the
+   * directory itself and the directory's parent directory are not included in the result. Each
+   * string is a path relative to the given directory.
+   *
+   * <p>
+   * There is no guarantee that the name strings in the resulting array will appear in any specific
+   * order; they are not, in particular, guaranteed to appear in alphabetical order.
+   *
+   * @param path the abstract pathname to list
+   * @param options for list directory
+   * @return An array of statuses naming the files and directories in the directory denoted by this
+   *         abstract pathname. The array will be empty if the directory is empty. Returns
+   *         {@code Optional.empty()} if this abstract pathname does not denote a directory.
+   */
+  default Optional<UfsStatus[]> listStatuses(String path, ListOptions options) throws IOException {
```

Comment:
add this function to UFS since I think it's a common need. @LuQQiu @huanght1997

Reply:
You introduced a new function to handle the case of listStatus() of a file.
I think the reason is to keep backward compatibility of the code, because the `listStatuses()` will only be used by  Job/Scheduler?

Reply:
If so, this function looks fine.

Reply:
yes, mainly for backward compatibility. We later on should consolidate those API